sourcegraph/syntect_server
# various updates, remove non-OSS grammars

- Update syntect version to latest `master` branch:
    - Changes: https://github.com/slimsag/syntect/compare/0a78296...slimsag:c1252e7
    - Upstream versions included: 4.2.0, 4.3.0, 4.4.0 ([changelog](https://github.com/trishume/syntect/blob/master/CHANGELOG.md))
- Update slimsag/Packages version to latest `master` which removes non-OSS and non-commercial grammars:
    - Perforce (GPL)
    - TOML (no license)
    - INI (no license)
    - GraphQL (no license)
    - Note: Please see https://github.com/sourcegraph/sourcegraph/issues/13933 for why we're removing these and how we hope to get support for these languages back again.
- Compile using Rust stable.
- Use Rust edition 2018 (latest)
- Use unpublished version of Rocket (0.5.0 is not released yet, required for compiling with Rust stable.)
- Refactor code for Rust async/await changes so it actually compiles.
- Release v1.0.0
- Remove onig_sys patch (previously required for cross-compilation.)

For a detailed walk-through of how painful this upgrade was for me to perform (relatively painless compared to past upgrades), see:

https://gist.github.com/slimsag/19e40ad3c0fcdc60a43674d1083a474c

Helps https://github.com/sourcegraph/sourcegraph/issues/13933
